Add new module cs_storage_ip_range

netservers opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Add a new CloudStack module for adding/updating/removing Storage IP Ranges.

This is part of a set of new modules that will also allow management of:
- Physical Networks
- TrafficTypes
- Storage IP Ranges
- Primary Storage Pools
- VLAN IP Ranges
- Console SSL Certificates
<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New Module Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
lib/ansible/modules/cloud/cloudstack/cs_storage_ip_range.py
##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible 2.4.0
  config file = /home/john/.ansible.cfg
  configured module search path = [u'/home/john/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible-2.4.0-py2.7.egg/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.13 (default, May 10 2017, 20:04:28) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]

```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->
Sample playbook:
<!--- Paste verbatim command output below, e.g. before and after your change -->
```
---
- name: Setup Cloudstack
  hosts: host1
  tasks:
  - name: Create Public VLAN IP Range
    local_action:
      module: cs_storage_ip_range
      pod: Pod01
      ipv4_start: 1.0.0.10
      ipv4_end: 1.0.0.100
      ipv4_netmask: 255.255.255.0
      ipv4_gateway: 1.0.0.1
      vlan: 101
```

Commit:
Add new module cs_storage_ip_range

gundalow commented:
/cc @resmo FYI

resmo commented:
waiting for integration tests

resmo commented on lib/ansible/modules/cloud/cloudstack/cs_storage_ip_range.py:
```diff
@@ -0,0 +1,305 @@
+def main():
+    argument_spec = cs_argument_spec()
+    argument_spec.update(dict(
+        pod=dict(default=None),
```
missing required=True here?

netservers replied:
Yes. Has now been corrected.

resmo commented on lib/ansible/modules/cloud/cloudstack/cs_storage_ip_range.py:
```diff
@@ -0,0 +1,305 @@
+)
+
+
+class AnsibleCloudStackCluster(AnsibleCloudStack):
```
Would you mind choosing a more accourate name, e.g. AnsibleCloudStackStorageIpRange?

netservers replied:
Done. Was just copying and pasting. I'll have to review my other modules too.

resmo commented on lib/ansible/modules/cloud/cloudstack/cs_storage_ip_range.py:
```diff
@@ -0,0 +1,305 @@
+        pods = self.cs.listPods(**args)
+        if pods:
+            return self._get_by_key(key, pods['pod'][0])
+        self.module.fail_json(msg="Pod %s not found in zone %s." % (self.module.params.get('pod'), self.get_zone(key='name')))
```
just a little hint, there is a cloudstack optimized fail_json `self.fail_json` which would also return the zone on failure as return value. But this is fine as well

netservers replied:
OK, I'll try to remember that.

resmo commented on lib/ansible/modules/cloud/cloudstack/cs_storage_ip_range.py:
```diff
@@ -0,0 +1,305 @@
+    argument_spec = cs_argument_spec()
+    argument_spec.update(dict(
+        pod=dict(default=None),
+        vlan=dict(default=None),
```
I would suggest to set `type='int' for vlan.

netservers replied:
This is also done.

resmo commented on lib/ansible/modules/cloud/cloudstack/cs_storage_ip_range.py:
```diff
@@ -0,0 +1,305 @@
+  returned: success
+  type: string
+  sample: a3fca65a-7db1-4891-b97c-48806a978a96
+zoneid:
```
`zone` is returned, but not zoneid

netservers replied:
Removed

resmo commented on lib/ansible/modules/cloud/cloudstack/cs_storage_ip_range.py:
```diff
@@ -0,0 +1,305 @@
+  returned: success
+  type: string
+  sample: a3fca65a-7db1-4891-b97c-48806a978a96
+podid:
```
I don't see code returning podid, I usually do not return the ids for the arguments. When you looks at the integration tests, you'll see I compare the return values with the arguments to make sure the API used the same. I see no benefit in returning id for subelements.

netservers replied:
I made the mistake of copying down the list of variables the API documentation claims to return. I've now shortened the list to those I actually see returned.

Commit:
Remove documentation about data not returned by the API.
Change class name.
Change "pod" to a required parameter.
Set 'int' datatype for vlan.

resmo commented:
in my queue for testing

resmo commented:
module still needs integration tests. If the author is not picking it up, I will work on it for ansible 2.5. 

waiting_on_contributor

resmo commented:
needs_revision

resmo commented:
close_me
